job/job.go

```diff
@@ -0,0 +1,188 @@
+package job
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type Job struct {
+	Id string `json:"id"`
+}
+
+type Status struct {
+	Status   string `json:"status"`
+	Location string `json:"location"` // Location is the unauthenticated temporary (60 secs) url used to download data
+}
+
+func CreateJob(url, authToken string, jobConfig string) (*Job, error) {
+
+	JobResponse := Job{}
+
+	payload := strings.NewReader(jobConfig)
+
+	req, _ := http.NewRequest("POST", url, payload)
+
+	req.Header.Add("authorization", fmt.Sprintf("Bearer %s", authToken))
+	req.Header.Add("content-type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if res.StatusCode != http.StatusOK {
```

Comment:
I *think* we should check the error before assuming anything about the response.
Don't think connection errors etc are checked in this case.